process/numShardsProcessor.go

```diff
@@ -0,0 +1,147 @@
+package process
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+)
+
+var errTimeIsOut = errors.New("time is out")
+
+const (
+	networkConfigPath = "/network/config"
+)
+
+type networkConfigResponseData struct {
+	Config struct {
+		NumShards uint32 `json:"erd_num_shards_without_meta"`
+	} `json:"config"`
+}
+
+type networkConfigResponse struct {
+	Data  networkConfigResponseData `json:"data"`
+	Error string                    `json:"error"`
+	Code  string                    `json:"code"`
+}
+
+// ArgNumShardsProcessor is the DTO used to create a new instance of numShardsProcessor
+type ArgNumShardsProcessor struct {
+	HttpClient                    HttpClient
+	Observers                     []string
+	TimeBetweenNodesRequestsInSec int
+	NumShardsTimeoutInSec         int
+	RequestTimeoutInSec           int
+}
+
+type numShardsProcessor struct {
+	observers                []string
+	httpClient               HttpClient
+	timeBetweenNodesRequests time.Duration
+	numShardsTimeout         time.Duration
+	requestTimeout           time.Duration
+}
+
+// NewNumShardsProcessor returns a new instance of numShardsProcessor
+func NewNumShardsProcessor(args ArgNumShardsProcessor) (*numShardsProcessor, error) {
```

Comment:
maybe we can use this component as a network config provider. there are also other useful information in the endpoint (chain id, min gas limit, max gas limit per tx, and so on) that can be used in other places

Reply:
for example, the faucet processor fetches the network config from observers at each request and this is not needed. base processor could hold an instance of the network config provider. or simply a network config structure

Reply:
as discussed, this implies caching the config, which might get changed after upgrade

we can keep this component, as it is only used once during startup